src/crimson/os/seastore/segment_cleaner.cc

```diff
@@ -423,6 +427,41 @@ SegmentCleaner::gc_reclaim_space_ret SegmentCleaner::gc_reclaim_space()
                   "SegmentCleaner::gc_reclaim_space: addr {} alive, gc'ing {}",
                   addr,
                   *ext);
+		assert(commit_time);
+		assert(commit_type == record_commit_type_t::MODIFY
+		  || commit_type == record_commit_type_t::REWRITE);
+		if (ext->get_last_modified() ==
+		    seastar::lowres_system_clock::time_point()) {
+		  assert(ext->get_last_rewritten() ==
+		    seastar::lowres_system_clock::time_point());
+		  ext->set_last_modified(
+		    seastar::lowres_system_clock::duration(
+		      commit_time));
```

Comment:
Should we add an assert here that `commit_type == record_commit_type_t::MODIFY`, as this is setting the last-modified to the extent?

Reply:
Um, this is a mistake, extents with zero `last_modified` value can also be met when scanning REWRITE commits, and I shouldn't update those extents' `last_modifiied` with REWRITE commits' `commit_time`.

I've corrected this part of the code, that is keep extents' `last_modified` time in `extent_info_t` and using these values to update extents with zero `last_modified` value.